editor/editor_properties_array_dict.cpp

```diff
@@ -504,6 +505,26 @@ bool EditorPropertyArray::_is_drop_valid(const Dictionary &p_drag_data) const {
 		return true;
 	}
 
+	if (drop_type == "nodes") {
+		Array node_paths = drag_data["nodes"];
+
+		if (allowed_type == "NodePath") {
```

Comment:
```cpp
	if (drop_type == "nodes") {
		Array node_paths = drag_data["nodes"];

		PackedStringArray allowed_subtype_array;
		if (allowed_type == "NodePath") {
			if (subtype_hint_string == "NodePath") {
				return true;
			} else {
				for (int j = 0; j < subtype_hint_string.get_slice_count(","); j++) {
					String ast = subtype_hint_string.get_slice(",", j).strip_edges();
					allowed_subtype_array.append(ast);
				}
			}
		}

		bool is_drop_allowed = true;
		for (int i = 0; i < node_paths.size(); i++) {
			Node *dropped_node = get_node_or_null(node_paths[i]);
			ERR_FAIL_NULL_V_MSG(dropped_node, false, "Could not get the dropped node by its path.");

			if (allowed_type == "NodePath") {
				if (subtype_hint_string != "NodePath") { // The array of NodePaths is restricted to specific types using @export_node_path()
					if (allowed_subtype_array.has(dropped_node->get_class_name())) {
						continue;
					} else { // The dropped node type was not found in the allowed subtype array, we must check if it does not inherit one of them
						for (String ast : allowed_subtype_array) {
							if (!ClassDB::is_parent_class(dropped_node->get_class_name(), ast)) {
								is_drop_allowed = false;
							} else {
								is_drop_allowed = true;
								break;
							}
						}
					}
					return is_drop_allowed;
				}
			}
			
			if (!ClassDB::is_parent_class(dropped_node->get_class_name(), allowed_type)) {
				// Fail if one of the nodes is not of allowed type.
				return false;
			}
		}

		// If no nodes fail, drop is valid.
		return true;
	}
```

I've done it like that. It works, but before pushing it in this form; do you have any suggestions ?

Reply:
I think the general idea is fine (assuming it works). If you push it I will add style suggestions.

Reply:
Never mind it was almost working but there's an issue I just found out :( ... I will push again when I fixed it.

Reply:
What issue?

Reply:
The issue was that if a node with a type that is not specified nor inheriting one of them was after one that was inheriting, it would have been allowed to be dropped. But with the changes I just pushed this is fixed.